storage/ndb/rest-server/rest-api-server/internal/integrationtests/benchmark_util.go

```diff
@@ -0,0 +1,103 @@
+/*
+ * This file is part of the RonDB REST API Server
+ * Copyright (c) 2023 Hopsworks AB
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, version 3.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package integrationtests
+
+import (
+	"math/rand"
+	"net/http"
+	"runtime"
+	"sort"
+	"strings"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"hopsworks.ai/rdrs/internal/config"
+	"hopsworks.ai/rdrs/internal/log"
+	"hopsworks.ai/rdrs/internal/testutils"
+)
+
+func RunRestTemplate(b *testing.B, method, url string, reqs *[]string, numRequests int) {
+	var nReq = len(*reqs)
+	maxProcs := runtime.GOMAXPROCS(0)
+	log.Infof("GOMAXPROCS: %d", maxProcs)
+	b.N = numRequests
+
+	latenciesChannel := make(chan time.Duration, numRequests)
+	var threadId int32 = 0
+	var numError int32 = 0
+	conf := config.GetAll()
+	log.Infof("Starting benchmark test in parallel")
+	b.ResetTimer()
+	start := time.Now()
+
+	b.RunParallel(func(bp *testing.PB) {
+		atomic.AddInt32(&threadId, 1)
+		var httpClient *http.Client = testutils.SetupHttpClient(b)
+
+		for bp.Next() {
+			var req, _ = http.NewRequest(
+				method,
+				url,
+				strings.NewReader((*reqs)[rand.Intn(nReq)]))
+			req.Header.Set("Content-Type", "application/json")
+			if conf.Security.APIKey.UseHopsworksAPIKeys {
+				req.Header.Set(config.API_KEY_NAME, testutils.HOPSWORKS_TEST_API_KEY)
+			}
+			requestStartTime := time.Now()
+			resp, err := httpClient.Do(req)
+			latenciesChannel <- time.Since(requestStartTime)
+			if resp == nil || resp.StatusCode != 200 {
+				atomic.AddInt32(&numError, 1)
+				log.Infof("Error status code: %d", resp.StatusCode)
+				if err != nil {
+					log.Info(err.Error())
+				} else {
+					var respStr []byte = make([]byte, 1000)
```

Comment:
what would happen if the response is more than 1000 bytes

Reply:
The response will be truncated. but it is just info message.